scripts/validate_unwanted_patterns.py

```diff
@@ -102,6 +103,33 @@ def bare_pytest_raises(file_obj: IO[str]) -> Iterable[Tuple[int, str]]:
                 break
 
 
+def private_function_across_module(file_obj: IO[str]) -> Iterable[Tuple[int, str]]:
+    """
+    Checking that a private function is not used across modules.
+
+    Parameters
+    ----------
+    file_obj : IO
+        File-like object containing the Python code to validate.
+
+    Yields
+    ------
+    line_number : int
+        Line number of import statement, that imports the private function.
+    msg : str
+        Explenation of the error.
+    """
+    contents = file_obj.read()
+    tree = ast.parse(contents)
+
+    for node in ast.walk(tree):
+        if not (isinstance(node, ast.Import) or isinstance(node, ast.ImportFrom)):
+            continue
+
+        if any(mod.name.split(".")[-1].startswith("_") for mod in node.names):
+            yield (node.lineno, "Use of private function across modules found.")
```

Comment:
The logic behind this is as follow:

1. If the node is not ```import``` or ```from foo import bar``` we continue to loop.
2. if the imported function name starts with an underscore(```_```), it will yield the line number of the import statement.

Reply:
can you add the function name to the message

Reply:
Sure!

Since 4786cf7 the private function imported name is included in the message.

Reply:
this is great, thanks